### DISCUSSION TITLE
Why ''Abort()'' doesn't work

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1504] [Q&A] [answered]
Why does `Abort(MapFixes.dll)` not work in this case? [log](https://github.com/user-attachments/files/17263122/ScriptHookVDotNet.log)
![image](https://github.com/user-attachments/assets/e7ed7b2c-1f20-4a1c-a3f2-3624a1ad0c4b)

**Comment by kagikn** [comment]
Because `Abort(string)` expects a string but what you passed can't be a string. What you passed was interpreted as the namespace named `MapFixes` and the name after the namespace couldn't expect a variable.

You want to surround a dll name with double quotes so the name will be interpreted as a literal string in this case, like `"MapFixes.dll"`.

**Reply by Michael21107** [reply]
thanks, seems to work even without the `.dll` in the command, i have 3 questions tho:
1. how to write `'` in the console since it seems like my classic shortcut doesn't work there (ALT+P)?
2. would i also have a CS script with the same name, how would SVHDN know which one i want to abort?
3. is there a way to abort a script without `'`?

**Reply by kagikn** [reply, marked as answer]
answers:
1. you press the key that says "2" while holding a Shift. You're forced to use the keyboard with the US layout (or maybe UK layout if already specified), right?
2. i didn't know this, but SHVDN use `.dll` as the specified extension when you don't specify anything as the extension, as these lines say.
https://github.com/scripthookvdotnet/scripthookvdotnet/blob/b5c7d931a5a42c68be2cacf487369ea996f18a1b/source/core/DllMain.cpp#L185-L186
specify extension to abort a `.cs` or `.vb` script.
3. no, you can't really do that without *double* quotes, not single quotes because single character with surrounding single quotes means a `char` literal (multiple characters with ones are invalid). Learn how C# works if you still doubt.
